oauth2_http/java/com/google/auth/oauth2/DownscopedCredentials.java

```diff
@@ -112,6 +112,18 @@ private DownscopedCredentials(Builder builder) {
     } else {
       this.universeDomain = builder.universeDomain;
     }
+
+    // Ensure source credential's universe domain matches.
+    try {
+      if (!this.universeDomain.equals(sourceCredential.getUniverseDomain())) {
+        throw new IllegalArgumentException(
+            "The downscoped credential's universe domain must be the same as the source "
+                + "credential.");
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException(
```

Comment:
nit: is it possible to add the IOException as the cause of the IllegalStateException here?